hoot-core/src/main/cpp/hoot/core/visitors/SetTagVisitor.cpp

```diff
@@ -72,7 +74,17 @@ void SetTagVisitor::visit(const shared_ptr<Element>& e)
   }
   else
   {
-    e->getTags()[_k] = _v;
+    if (e->getTags().keys().contains(_k) && _appendToExistingValue)
```

Comment:
Possibly swap the order of the conditions here - so that if we are not appending, we don't have to search for _k